src/embedded/index.ts

```diff
@@ -145,80 +151,173 @@ export class EmbeddedDB {
-        reject(
-          new Error(
-            `failed to download binary: not found. ` +
-              `are you sure Weaviate version ${opt.version} exists? ` +
-              `note that embedded db is only supported for versions >= 1.18.0`
-          )
-        );
-      } else {
-        reject(new Error(`failed to download binary: unexpected status code: ${resp.statusCode}`));
-      }
-    }).on('error', function (err) {
-      fs.unlinkSync(tarballPath);
-      reject(new Error(`failed to download binary: ${JSON.stringify(err)}`));
+        } else if (resp.statusCode == 404) {
+          reject(
+            new Error(
+              `failed to download binary: not found. ` +
+                `are you sure Weaviate version ${this.options.version} exists? ` +
+                `note that embedded db is only supported for versions >= 1.18.0`
+            )
+          );
+        } else {
+          reject(new Error(`failed to download binary: unexpected status code: ${resp.statusCode}`));
+        }
+      }).on('error', (err) => {
+        fs.unlinkSync(tarballPath);
+        reject(new Error(`failed to download binary: ${JSON.stringify(err)}`));
+      });
     });
-  });
-}
+  }
 
-function untarBinary(opt: EmbeddedOptions, tarballPath: string): Promise<null> {
-  const tarball = fs.createReadStream(tarballPath);
-  return new Promise((resolve, reject) => {
-    tarball.pipe(
-      extract({
-        cwd: dirname(tarballPath),
-        strict: true,
-      })
-        .on('finish', () => {
-          tarball.close();
-          fs.renameSync(join(dirname(opt.binaryPath), 'weaviate'), opt.binaryPath);
-          resolve(null);
+  private buildBinaryUrl(): string {
+    let arch: string;
+    switch (process.arch) {
+      case 'arm64':
+        arch = 'arm64';
+        break;
+      case 'x64':
+        arch = 'amd64';
+        break;
+      default:
+        throw new Error(`Embedded DB unsupported architecture: ${process.arch}`);
+    }
+
+    return (
+      `https://github.com/weaviate/weaviate/releases/download/v${this.options.version}` +
+      `/weaviate-v${this.options.version}-linux-${arch}.tar.gz`
+    );
+  }
+
+  private untarBinary(tarballPath: string): Promise<null> {
+    const tarball = fs.createReadStream(tarballPath);
+    return new Promise((resolve, reject) => {
+      tarball.pipe(
+        extract({
+          cwd: dirname(tarballPath),
+          strict: true,
         })
-        .on('error', function (err) {
-          reject(new Error(`failed to untar binary: ${JSON.stringify(err)}`));
+          .on('finish', () => {
+            tarball.close();
+            fs.renameSync(join(dirname(this.options.binaryPath), 'weaviate'), this.options.binaryPath);
+            resolve(null);
+          })
+          .on('error', (err) => {
+            reject(new Error(`failed to untar binary: ${JSON.stringify(err)}`));
+          })
+      );
+    });
+  }
+
+  private waitTillListening(): Promise<null> {
```

Comment:
just a note, this function could be used to implement [this here](https://github.com/weaviate/typescript-client/issues/17)
(obviously should not be part of this PR)